packages/block-library/src/navigation/overlay.scss

```diff
@@ -0,0 +1,223 @@
+// ********************************************************
+// VARIABLES & MIXINS
+// ********************************************************
+
+// Abstraction for Navigation breakpoint.
+@mixin nav-breakpoint() {
+	@include break-small {
+		@content;
+	}
+}
+
+
+// ********************************************************
+// ANIMATION
+// ********************************************************
+@keyframes overlay-menu__fade-in-animation {
+	from {
+		opacity: 0;
+		transform: translateY(0.5em);
+	}
+
+	to {
+		opacity: 1;
+		transform: translateY(0);
+	}
+}
+
+
+// ********************************************************
+// ROOT
+// Alias: wp-block-navigation__responsive-container
+// ********************************************************
+.wp-overlay-component {
+	display: none;
+	position: fixed;
+	top: 0;
+	left: 0;
+	right: 0;
+	bottom: 0;
+
+	@include nav-breakpoint {
+		&:not(.hidden-by-default):not(.is-overlay-open) {
+			display: block;
+			width: 100%;
+			position: relative;
+			z-index: auto;
+			background-color: inherit;
+		}
+	}
```

Comment:
I assume this code exists so that if the overlay is open and the screen size changes then the overlay becomes hidden?

I'm wondering. If in the future we want to allow the Nav breakpoint to change _dynamically_ based on measuring the width of the Navigation block, then we might like to look at moving all the responsive behaviour to JavaScript with something like `matchMedia` similar to what I did in https://github.com/WordPress/gutenberg/pull/45274.

The only issue is then we'd need to find a way to do a similar thing with directives on the front of the site.

I believe taking such an approach would make this code a lot simpler.

Reply:
yes, I was thinking on using a class that will be added or removed once we reach the correct viewport width